File: python/1.py
 #https://leetcode.com/problems/two-sum/submissions/
 # very easy two sum and it test hash table, itetation, slide window
 class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        h = dict()
        for i,v in enumerate(nums):
            if target - v in h:
                return [h[target - v], i]
            else:
                h[v ] = i

File: go/1.go
func twoSum(nums []int, target int) []int {
    
    reserve := make(map[int] int)
    for i,v := range nums {
        if val, ok:= reserve[target - v]; ok{
            return []int{i, val}
        }
        reserve[v] = i
        
    }
    return []int{-1, -1}
    
}
